lib/compiler.js

```diff
@@ -38,6 +38,8 @@ module.exports = function compile(options) {
 
   var appRootDir = options.appRootDir = options.appRootDir || process.cwd();
   var env = options.env || process.env.NODE_ENV || 'development';
+  var scriptExtensions = options.scriptExtensions = options.scriptExtensions ||
```

Comment:
Modifying input arguments (setting `options.scriptExtensions`) is a bad practice, we have been bitten by this in the past. Please treat `options` are readonly/frozen.

I see that you are accessing `options.scriptExtensions` at https://github.com/strongloop/loopback-boot/pull/242/files#diff-ee97c5091b89979aace94674818996baR722.

Here are two of many possible solutions how to keep `options` frozen and still have `scriptExtensions` set:

```js
// use prototypal inheritance
options = Object.create(options);

// shallow-copy properties
options = util._extend({}, options);
```

(A line like the above would go to L38-39 above, before we start processing the `options` object.)